client/layout/guided-tours/config-elements/step.tsx

```diff
@@ -94,37 +94,42 @@ export default class Step extends Component< Props, State > {
 	 */
 	isUpdatingPosition: boolean = false;
 
-	componentDidMount() {
-		this.wait( this.props, this.context ).then( () => {
-			this.start();
-			this.setStepSection( this.context, { init: true } );
-			debug( 'Step#componentDidMount: stepSection:', this.stepSection );
-			this.skipIfInvalidContext( this.props, this.context );
-			this.scrollContainer = query( this.props.scrollContainer )[ 0 ] || window;
-			this.setStepPosition( this.props );
-			this.setState( { initialized: true } );
-			window.addEventListener( 'resize', this.onScrollOrResize );
-			this.watchTarget();
-		} );
+	async componentDidMount() {
```

Comment:
just to make sure the intent is correct here, react doesn't support this being an async function in any special way. All this really adds is the ability to use `await` inside the function. It does not delay future invocations of other lifecycle methods.

Reply:
Yes :+1: This is reflected by the use of [this.setState( { initialized: true } );](https://github.com/Automattic/wp-calypso/pull/33409/files#diff-7a8a889ae1417e2659eb1b4270a5529aR110), which makes sure that `render()` [bails early](https://github.com/Automattic/wp-calypso/blob/9aaba11bab7cfee8b7057da2f42bde03615f6c45/client/layout/guided-tours/config-elements/step.tsx#L352-L354) if called before `this.wait()` has resolved.